src/Doctrine/Orm/Filter/AttributeFilter.php

```diff
@@ -0,0 +1,272 @@
+            $queryNameGenerator,
+            $alias,
+            $property,
+            $attributeId,
+            $extractedValue
+        );
+    }
+
+    /** @param mixed $value */
+    protected function addWhere(
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $alias,
+        string $field,
+        string $attributeId,
+        $value,
+        string $operator = self::OPERATOR_EXACT
+    ): void {
+        $valueParameter = $queryNameGenerator->generateParameterName($field);
+
+        /** @var AttributeInterface $attribute */
+        $attribute = $this->iriConverter->getItemFromIri($attributeId);
+        $attributeType = $attribute->getType();
+        Assert::notNull($attributeType);
+        $value = $this->normalizeValue($value, $attributeType);
+
+        switch ($operator) {
+            case self::OPERATOR_EXACT:
+                $queryBuilder
+                    ->andWhere(
+                        sprintf(
+                            '%s.%s = :%s',
+                            $alias,
+                            $attributeType,
+                            $valueParameter
+                        )
+                    )
+                    ->setParameter($valueParameter, $value);
+
+                break;
+            case self::OPERATOR_PARTIAL:
+                if (null === $value) {
+                    return;
+                }
+
+                $queryBuilder
+                    ->andWhere(sprintf('%s.%s > :%s', $alias, $field, $valueParameter))
+                    ->setParameter($valueParameter, $value);
+
+                break;
+        }
+    }
+
+    public function getDescription(string $resourceClass): array
+    {
+        $description = [];
+
+        $properties = $this->getProperties();
+        if (null === $properties) {
+            $properties = array_fill_keys($this->getClassMetadata($resourceClass)->getFieldNames(), null);
+        }
+
+        /** @var string $property */
+        foreach (array_keys($properties) as $property) {
+            $description += $this->getFilterDescription($property, self::ATTRIBUTE_ID);
+            $description += $this->getFilterDescription($property, self::VALUE);
+        }
+
+        return $description;
+    }
+
+    protected function getFilterDescription(string $fieldName, string $operator): array
+    {
+        /** @var string $propertyName */
+        $propertyName = $this->normalizePropertyName($fieldName);
+
+        return [
+            sprintf('%s[%s]', $propertyName, $operator) => [
+                'property' => $propertyName,
+                'type' => 'string',
+                'required' => false,
+            ],
+        ];
+    }
+
+    private function getAttributeId(array $values, string $property): ?string
+    {
+        if (array_key_exists(self::ATTRIBUTE_ID, $values)) {
+            /** @var string $attributeId */
+            $attributeId = $values[self::ATTRIBUTE_ID];
```

Comment:
Why not return it early and get rid of the `if-else` construct?

Reply:
You're right, I'm changing that.